tests/bluetooth/bsim_bt/bsim_test_audio/src/vcs_test.c

```diff
@@ -0,0 +1,754 @@
+/*
+ * Copyright (c) 2019 Bose Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BT_VCS
+#include "bluetooth/audio/vcs.h"
+#include "common.h"
+
+extern enum bst_result_t bst_result;
+
+#if defined(CONFIG_BT_VOCS)
+#define VOCS_DESC_SIZE CONFIG_BT_VOCS_MAX_OUTPUT_DESCRIPTION_SIZE
+#else
+#define VOCS_DESC_SIZE 0
+#endif /* CONFIG_BT_VOCS */
+
+#if defined(CONFIG_BT_AICS)
+#define AICS_DESC_SIZE CONFIG_BT_AICS_MAX_INPUT_DESCRIPTION_SIZE
+#else
+#define AICS_DESC_SIZE 0
+#endif /* CONFIG_BT_AICS */
+
+static struct bt_vcs vcs;
+
+static volatile uint8_t g_volume;
+static volatile uint8_t g_mute;
+static volatile uint8_t g_flags;
+static volatile int16_t g_vocs_offset;
+static volatile uint8_t g_vocs_location;
+static char g_vocs_desc[VOCS_DESC_SIZE];
+static volatile int8_t g_aics_gain;
+static volatile uint8_t g_aics_input_mute;
+static volatile uint8_t g_aics_mode;
+static volatile uint8_t g_aics_input_type;
+static volatile uint8_t g_aics_units;
+static volatile uint8_t g_aics_gain_max;
+static volatile uint8_t g_aics_gain_min;
+static volatile bool g_aics_active = 1;
+static char g_aics_desc[AICS_DESC_SIZE];
+static volatile bool g_cb;
+static struct bt_conn *g_conn;
+static bool g_is_connected;
+
+static void vcs_state_cb(struct bt_conn *conn, int err, uint8_t volume,
+			 uint8_t mute)
+{
+	if (err) {
+		FAIL("VCS state cb err (%d)", err);
+		return;
+	}
+
+	g_volume = volume;
+	g_mute = mute;
+
+	if (!conn) {
+		g_cb = true;
+	}
+}
+
+static void vcs_flags_cb(struct bt_conn *conn, int err, uint8_t flags)
+{
+	if (err) {
+		FAIL("VCS flags cb err (%d)", err);
+		return;
+	}
+
+	g_flags = flags;
+
+	if (!conn) {
+		g_cb = true;
+	}
+}
+
+static void vocs_state_cb(struct bt_conn *conn, struct bt_vocs *inst, int err,
+			  int16_t offset)
+{
+	if (err) {
+		FAIL("VOCS state cb err (%d)", err);
+		return;
+	}
+
+	g_vocs_offset = offset;
+
+	if (!conn) {
+		g_cb = true;
+	}
+}
+
+static void vocs_location_cb(struct bt_conn *conn, struct bt_vocs *inst,
+			     int err, uint32_t location)
+{
+	if (err) {
+		FAIL("VOCS location cb err (%d)", err);
+		return;
+	}
+
+	g_vocs_location = location;
+
+	if (!conn) {
+		g_cb = true;
+	}
+}
+
+static void vocs_description_cb(struct bt_conn *conn, struct bt_vocs *inst,
+				int err, char *description)
+{
+	if (err) {
+		FAIL("VOCS description cb err (%d)", err);
+		return;
+	}
+
+	strcpy(g_vocs_desc, description);
+
```

Comment:
In the client test, there is a size check.
(Not sure how important that is here.)

Reply:
Changed to use `strncpy`, but should not be necessary as we use the Kconfig that will have truncated the value from the service implementation anyhow :)